staging/src/k8s.io/apimachinery/pkg/runtime/extension.go

```diff
@@ -25,7 +26,10 @@ func (re *RawExtension) UnmarshalJSON(in []byte) error {
 	if re == nil {
 		return errors.New("runtime.RawExtension: UnmarshalJSON on nil pointer")
 	}
-	re.Raw = append(re.Raw[0:0], in...)
+
+	if !bytes.Equal(in, []byte("null")) {
+		re.Raw = append(re.Raw[0:0], in...)
```

Comment:
actually, this doesn't round trip correctly... it sets Raw to `[]byte{}`, which would not re-serialize to "null" but to an empty byte array, which would be invalid JSON. I think MarshalJSON has to change to `if len(re.Raw) == 0 {` or it risks outputting invalid JSON